WorkoutGeneratorApp/models/ExerciseModel.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable prettier/prettier */
+/* One set of an exercise, denoted by W x R. */
+interface ExerciseSet {
+  readonly Weight: number;
+  readonly Reps  : number;
+  IsCompleted    : boolean;
+  Type           : ExerciseType;
+  MuscleFocus    : MuscleGroup;
+  IntensityLevel : ExerciseIntensity;
+}
+
+/* Base Exercise model. */
+export interface Exercise {
+  readonly Name  : string;
+  readonly Sets  : ReadonlyArray<Omit<ExerciseSet, 'IsCompleted'>>;
+  readonly Unit  : MeasurementSystem;
+  readonly Image?: unknown; //TBD
+  readonly Media?: unknown; //TBD
+  RestTime?      : number;  //TBD, editable by user for long rest periods?
+}
```

Comment:
Not sure what the system will look like yet, but do you think we can add a member for MuscleGroup here as well?

Reply:
Yes, I think that was what I intended to do, thanks.

Reply:
Im thinking we should separate the sets from the exercise because the exercise itself is independent of the sets. But its also not a big deal and Im not sure if this change will be beneficial in our current use case, Though it may be in the future.